internal/osbuild2/dnf_config_stage.go

```diff
@@ -1,22 +1,49 @@
 package osbuild2
 
+import (
+	"fmt"
+)
+
 // DNFConfigStageOptions represents persistent DNF configuration.
 type DNFConfigStageOptions struct {
 	// List of DNF variables.
 	Variables []DNFVariable `json:"variables,omitempty"`
+	Config    *DNFConfig    `json:"config,omitempty"`
 }
 
 func (DNFConfigStageOptions) isStageOptions() {}
 
 // NewDNFConfigStageOptions creates a new DNFConfig Stage options object.
-func NewDNFConfigStageOptions(variables []DNFVariable) *DNFConfigStageOptions {
+func NewDNFConfigStageOptions(variables []DNFVariable, config *DNFConfig) *DNFConfigStageOptions {
 	return &DNFConfigStageOptions{
 		Variables: variables,
+		Config:    config,
+	}
+}
+
+func (o DNFConfigStageOptions) validate() error {
+	if o.Config != nil && o.Config.Main != nil {
+		valid := false
+		allowedIPR := []string{"4", "IPv4", "6", "IPv6"}
+		for _, v := range allowedIPR {
+			if o.Config.Main.IPResolve == v {
```

Comment:
This is problematic. The structure member is of type `string`. If we ever extend the stage to allow more options in the `main` section, one can decide to set that option, but not the `IPResolve` option. In such case its value will be empty string. This check will always fail on the default string type value and the produced error message will be a bit misleading. In addition, the stage schema does not prevent specifying empty `config` and `main` properties.

You could probably add `""` to the list of valid values or make the member a pointer to `string`. The second option probably does not make much sense, since the empty string is not a valid stage schema value so we can use it as indication that the value was not set..